### DISCUSSION TITLE
Little idea of enhancement

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/grandorgue#806] [General]
Hello everyone,
I just have a little idea of ​​improvement of GO.

What would you think that on the FIRST start of GO (just after installation), the program asks the user to save the hardware configuration of his organ (number and extent of keyboards, pedalboards, expressive boxes, buttons etc. )

Thus, the user will not need each organ installed to have to redo this configuration.
simply, he will only have to assign (by selection for example) the elements of the installed organ to his own configuration.

For example,

- I have a 2-key organ and I want to install a 3-key instrument.
My equipment is already recognized and by selection, I can assign this or that keyboard where I want (eg GO: 2nd keyboard, Positive: 1st keyboard etc ..)
GO shows a warning because of the difference of keyboards (I can assign 2 keyboards on 1 or lose a keyboard)

- another example I have a 2 keyboards organ and I want to install a 2 keyboards organ: everything is automatically recognized without having to reconfigure the recognition of the hardware elements.
I can however change the order of the keyboards by a simple selector button a posteriori of the installation of the organ.

I don't know if it's very clear ...

**Comment by oleg68** [comment]
@Jean-Gabriel1982 It is already possible in `File->Settings->Initial Midi Configuration` with the current version of GO.

**Reply by Jean-Gabriel1982** [reply]
@oleg68 thank you very much, indeed almost everything is there ... but without wanting to make my case a general ;-), I do not think that the majority of users are aware of this very practical possibility of configuring their equipment?

maybe we can consider a kind of "tutorial" when you first open GO?
